youtube_dl/extractor/brightcove.py

```diff
@@ -508,6 +510,20 @@ def _extract_urls(webpage):
                 'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
                 % (account_id, player_id, embed, video_id))
 
+        # 	<video data-video-id="5320421710001" data-account="245991542" data-player="SJWAiyYWg" data-embed="default" class="video-js" controls itemscope itemtype="http://schema.org/VideoObject">
+        for video in re.findall(r'(?i)(<video[^>]+>)', webpage):
+            attrs = extract_attributes(video)
+                
+            video_id    = attrs.get('data-video-id')
+            account_id  = attrs.get('data-account')
+            player_id   = attrs.get('data-player')
+            embed       = attrs.get('data-embed')
+
+            if video_id and account_id and player_id and embed:
+                entries.append(
+                    'http://players.brightcove.net/%s/%s_%s/index.html?videoId=%s'
+                    % (account_id, player_id, embed, video_id))
+
```

Comment:
As remitamine has said, there's a similar detection at https://github.com/rg3/youtube-dl/blob/16e2c8f/youtube_dl/extractor/brightcove.py#L494-L509. It's better to merge them together. Specificially, the detection can be done in 2 steps:

1. Detect the Brightcove loader script. It can be either a <script> tag, as [recommended by Brightcove](https://support.brightcove.com/en/video-cloud/docs/dynamically-assigning-videos-player) or via [script.js](https://github.com/ded/script.js), just like the example in http://www.programme-tv.net/videos/extraits/81095-guillaume-canet-evoque-les-rumeurs-d-infidelite-de-marion-cotillard-avec-brad-pitt-dans-vivement-dimanche/:
```Javascript
$script([
            '//players.brightcove.net/800000640001/HyenP6LuN_default/index.min.js',
            '//players.brightcove.net/800000640001/Hyx5mvd4_default/index.min.js'
        ], 'brightcove')
```
2. Check <video> tags. Just like what Brightcove loader scripts do: (below is taken from http://players.brightcove.net/800000640001/Hyx5mvd4_default/index.min.js)
```Javascript
    i = document.querySelectorAll("video");
    for (k = 0; k < i.length; k++) {
        j = i[k];
        if (j.getAttribute("data-player") === "Hyx5mvd4" && j.getAttribute("data-embed") === "default") {
            if (/(?:^|\s)vjs-tech(?:\s|$)/.test(j.className)) {
                videojs.log("Ignoring already initialized player");
                continue
            }
            h(j)
        }
    }
```